src/pytest_plugins/consume/hive_simulators/engine/test_via_engine.py

```diff
@@ -91,3 +105,199 @@ def test_blockchain_via_engine(
                         assert (
                             forkchoice_response.payload_status.status == PayloadStatusEnum.VALID
                         ), f"unexpected status: {forkchoice_response}"
+
+
+def check_post(expected_post: Alloc, eth_rpc: EthRPC):
+    """Check the post state against the expected post state."""
```

Comment:
this looks like a common logic. can't we used the one that is already implemented?